soc/arm/nordic_nrf/nrf91/soc.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for Nordic Semiconductor nRF91 family processor
+ *
+ * This module provides routines to initialize and support board-level hardware
+ * for the Nordic Semiconductor nRF91 family processor.
+ */
+
+#include <kernel.h>
+#include <init.h>
+#include <cortex_m/exc.h>
+#include <nrfx.h>
+#include <soc/nrfx_coredep.h>
+
+#ifdef CONFIG_RUNTIME_NMI
+extern void _NmiInit(void);
+#define NMI_INIT() _NmiInit()
+#else
+#define NMI_INIT()
+#endif
+
+#if defined(CONFIG_SOC_NRF9160)
```

Comment:
Why is this necessary?

Reply:
This keeps consistency with nRF52 soc.c, where there are multiple SOC variants. In the future we might have multiple nRF91 SOC variants, as well.